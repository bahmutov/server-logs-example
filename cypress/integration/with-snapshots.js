// loads Cypress types + definition for custom "cy.api" command
/// <reference types="@bahmutov/cy-api" />

const pickImportantMessages = ({ body, messages }) => {
  const computations = Cypress._.filter(messages, {
    type: 'debug',
    namespace: 'compute'
  })
  return {
    result: body,
    backendMessages: Cypress._.map(computations, 'message')
  }
}

it('performs expected computations', () => {
  cy.api(
    {
      url: '/?a=2&b=3'
    },
    '2 + 3'
  )
    .then(pickImportantMessages)
    .toMatchSnapshot()

  cy.api(
    {
      url: '/?a=30&b=-1'
    },
    '30 - 1'
  )
    .then(pickImportantMessages)
    .toMatchSnapshot()

  cy.api(
    {
      url: '/?a=+1&b=+2'
    },
    '+1 + 2'
  )
    .then(pickImportantMessages)
    .toMatchSnapshot()
})

it('computes all given examples', () => {
  const list = []
  const addToList = list.push.bind(list)
  ;[
    // all examples to compute
    '/?a=2&b=3',
    '/?a=30&b=-1',
    '/?a=+1&b=+2'
  ].forEach(url => {
    cy.api({ url })
      .then(pickImportantMessages)
      .then(addToList)
  })
  cy.wrap(list).toMatchSnapshot()
})
