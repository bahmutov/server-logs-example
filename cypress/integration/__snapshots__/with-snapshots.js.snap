exports[`performs expected computations #0`] =
{
  "backendMessages": [
    "2 + 3 = 5"
  ],
  "result": "5"
};

exports[`performs expected computations #1`] =
{
  "backendMessages": [
    "30 + -1 = 29"
  ],
  "result": "29"
};

exports[`performs expected computations #2`] =
{
  "backendMessages": [
    "1 + 2 = 3"
  ],
  "result": "3"
};

exports[`computes all given examples #0`] =
[
  {
    "backendMessages": [
      "2 + 3 = 5"
    ],
    "result": "5"
  },
  {
    "backendMessages": [
      "30 + -1 = 29"
    ],
    "result": "29"
  },
  {
    "backendMessages": [
      "1 + 2 = 3"
    ],
    "result": "3"
  }
];
